src/public/videoApp.ts

```diff
@@ -0,0 +1,154 @@
+import { sendMessageToParent } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { FrameContexts } from './constants';
+import { registerHandler } from '../internal/handlers';
+/**
+ * Namespace to video extensibility of the SDK.
+ *
+ */
+export namespace videoApp {
+  /**
+   * Represents a video frame.
+   */
+  export interface VideoFrame {
+    /**
+     * Video frame width.
+     */
+    width: number;
+    /**
+     * Video frame height.
+     */
+    height: number;
+    /**
+     * Video frame buffer
+     */
+    data: Uint8ClampedArray;
+    /**
+     * NV12 luma stride, valid only when video frame format is NV12
+     */
+    lumaStride: number;
+    /**
+     * NV12 chroma stride, valid only when video frame format is NV12
```

Comment:
do we need to make these fields optional?

Reply:
Yes, the fields are optional and we need to make they are optional too.